### ISSUE TITLE
Notify soundcraft-utils package maintainers of the rename and move (after 0.5 release)

### ISSUE THREAD

**Issue by ndim** [socratools/socranop#15]
After the socranop-0.5 release, we should notify the people involved in packaging `soundcraft-utils` of the rename and move to `socranop`, so they can take the required steps:

  * @pakettiale for NixOS
  * @udaemon for Arch Linux

And for completeness' sake, I @ndim have a package prepared for Fedora which I will submit for inclusion into Fedora once socranop-0.5 has been released.

**Comment by lack** [comment]
Included with this issue should be creating proper deprecation notices for the soundcraft-utils git page, repo README.md, and pypi package.

**Comment by matthewarmand** [comment]
@udaemon I've created an AUR package for `socranop` (see https://github.com/socratools/socranop/issues/63) and have submitted a deletion request on the AUR for `soundcraft-utils`

**Comment by matthewarmand** [comment]
> I could delete it, but then it just disappears and no hint to socratools will be left. I could leave it for a while and point to here in a comment.

There's been a comment there since July pointing to this issue, and one there from me since September pointing to the `socranop` package. If we think that's not sufficient, you could push a re-packaging version to `soundcraft-utils` that `echo`s a message in the install script on install and update to alert to the deprecation and pointing to `socranop` instead. I'm assuming as maintainer and last packager you're alerted to the deletion request, so if you wanted you could implement that warning and hold up the deletion for a period of time before actually deleting the package.

**Comment by ndim** [comment]
Uhm... messages to users on install or updates? Does Arch not have a way to rename a package?

If I had to supersede an old `soundcraft-utils` package with a `socranop` package on Fedora, I would have the `socranop` package "provide" `soundcraft-utils` with a higher version than the last `soundcraft-utils` package, and "obsolete" all `soundcraft-utils` versions before that.

That would ensure that every system with a `soundcraft-utils` package installed would just update from `soundcraft-utils` to `socranop` the next time they run a general package update. No user interaction needed.

It would surprise me if Arch does not have a comparable mechanism to rename/supersede a package.

**Comment by matthewarmand** [comment]
I mentioned in #63 that I'd considered having it `conflict`. There's also `replaces`, which has some guidelines in the AUR Packaging guidelines on the wiki, quoted below: https://wiki.archlinux.org/title/AUR_submission_guidelines#Rules_of_submission

> Do not use replaces in an AUR PKGBUILD unless the package is to be renamed, for example when Ethereal became Wireshark. If the package is an alternate version of an already existing package, use conflicts (and provides if that package is required by others). The main difference is: after syncing (-Sy) pacman immediately wants to replace an installed, 'offending' package upon encountering a package with the matching replaces anywhere in its repositories; conflicts, on the other hand, is only evaluated when actually installing the package, which is usually the desired behavior because it is less invasive.

It's not a hill I'll die on, but there seems to me to be more dividing `soundcraft-utils` and `socranop` than a simple rename.

The only real way to "rename" AUR packages would I suppose be a `merge`, which transfers votes and comments to the new package. But I think in order to get the automated replacing behavior we'd have to use `replaces` as above.

**Comment by ndim** [comment]
Short history of socranop:

@lack wrote `soundcraft-utils`. I joined to help improve system integration. We noticed we should better call it something that does not have "Soundcraft" as part of its name, so we called it `socranop` and gave it its own GitHub project https://github.com/socratools.

To me, that sounds a lot like Ethereal changing its name to Wireshark.

**Comment by matthewarmand** [comment]
All right, if that's how you want it to work I can make a change to the `socranop` package so that it `replaces` `soundcraft-utils`. I want to minimize pushes and rebuilds due to non-crucial packaging updates and would generally rather wait til the next `socranop` release. But this will help migration off the old package, so if there's a desire to get that rolling I can push it sooner.

We can also submit a merge request; given there's not many comments or votes it won't make a huge difference, but for the sake of completeness.

**Comment by matthewarmand** [comment]
Actually in light of this issue wanting to do this with the 0.5 release, let me put a fine point on that question... @ndim do you want this change now or should we hold til 0.5?

**Comment by ndim** [comment]
@matthewarmand This is not an urgent security issue at all. 

I would consider `socranop-0.5` to be the next release of `soundcraft-utils` — a release which happens to decrease the potential security impact by not running anything as root.

How to map that into distribution packages is mainly the respective distribution's area of responsibility.

I cowardly avoided this issue on Fedora by not packaging `soundcraft-utils` for Fedora at all up to now. I would just guess that providing an upgrade path which to users feels like any other update would make sense on any distribution.

I will not take another two years again with socranop 0.5. Hoping for something more like two weeks for my part than for two years.

**Comment by udaemon** [comment]
hi, i saw your posts here, but I'm not sure what to do now with the package. ha: even I don't understand if I as maintainer can delete it from AUR or if deletion has to be requested at, say, a "higher" instance, whoever. when logged in into my AUR-account I can't find an option to delete. I just see there's pending request, but neither can see it's content nor can I do sth about it. 
What I can do is disown the package which maybe appropriate if I understand your second last post, @matthewarmand.
Any hint or advice is much recommended!
:)

**Comment by matthewarmand** [comment]
@udaemon yeah it's the Trusted Users who will actually address and take action on the request. We're going to disregard the deletion request (I've sent an update to disregard the deletion request, that should hit the list soon).

What we're going to do:
- As part of the 0.5 release, I'll edit `socranop`'s `PKGBUILD` so that it `replaces` `soundcraft-utils`. This should cause `paru -Syu` and similar to automatically replace any existing `soundcraft-utils` installs with `socranop`.
- Then we'll file a `merge` request to merge `soundcraft-utils` into `socranop` on the AUR.

If you want to disown the package and/or make me a maintainer or a co-maintainer on `soundcraft-utils` I'm fine with that (https://aur.archlinux.org/account/heylookmatt), and I can try and help with the whole migration. Otherwise I'll just contact you when we need to run the `merge`.

**Comment by udaemon** [comment]
@matthewarmand thanks for your explanations!
 I've just added you as co-maintainer. pls let me know if I can contribute somehow.
